### DISCUSSION TITLE
custom access

### DISCUSSION THREAD

**Discussion by mu-abbas** [supabase/supabase#22522] [Questions] [answered]
i want to know how to give access to read the data for only custom users, i tried to update the role but i can't do, i tried to access the user_metadata and save a property with a role name, and always fails, does anyone know how to do it? 

i need to give admins a read access for a specific table but other users not

**Comment by GaryAustin1** [comment, marked as answer]
This is a complicated process.
You can either use RLS with a jwt claim for admin, a  join (select) on a user/role table or with a Postgres role from the role claim in the jwt.

To make a Postgres role work you have to do several grants so that the REST API can assign the role.

JWT approach https://github.com/supabase-community/supabase-custom-claims
Auth Hooks doing the same thing: https://supabase.com/docs/guides/auth/auth-hooks

Here is a discussion on setting up Postgres roles: https://github.com/orgs/supabase/discussions/11948

**Reply by mu-abbas** [reply]
GaryAustin, thank you for your effort, i will read the articles that you have provided
